source/OptimaJet.DWKit.StarterApplication/Forms/Products/BaseProductForm.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Linq;
+using Optimajet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Repositories;
+using OptimaJet.DWKit.StarterApplication.Services;
+
+namespace OptimaJet.DWKit.StarterApplication.Forms.Products
+{
+    public class BaseProductForm : BaseForm
+    {
+        public BaseProductForm(
+            UserRepository userRepository,
+            ICurrentUserContext currentUserContext) : base(userRepository, currentUserContext)
+        {
+        }
+        protected void ValidateProduct(Project project, ProductDefinition productDefinition)
+        {
+            if ((project == null) || (productDefinition == null))
+            {
+                // Allowing it to return in these cases should cause the base to hit normal
+                // foreign key failures
+                return;
+            }
+            if ((project.Organization == null))
+            {
+                var message = $"The project for this product does not have a valid organization";
+                AddError(message);
+                return; // Can't proceed further without causing exceptions
+            }
+
+            if (!project.Organization.ProductDefinitionIds.Contains(productDefinition.Id))
+            {
+                var message = $"This product is not defined for the product's project organization";
```

Comment:
I don't understand this error message.  What this check seems to do is very that ProductDefinition is permitted for the Organization.

Reply:
No product definition I translated as the product is not defined for.  These are all one step removed relationships which made it difficult to figure out how to put this.  Would "The product definition for this product is not permitted for the product's project organization?" be acceptable?

Reply:
How about: "This product is not permitted for this organization."

Reply:
How about "This product is not permitted for the product's project organization"